apps/frontend/src/app/(app)/space/reading-list/page.tsx

```diff
@@ -55,11 +55,13 @@ const ReadingListPage: React.FC = () => {
 
   return (
     <section className="h-full overflow-y-auto bg-background text-secondary-foreground">
-      <div className="max-w-[800px] mx-auto px-4 py-16 sm:px-6 lg:px-8">
-        <div className="flex flex-col gap-8 text-base">
-          <AddItemForm blockId={blockId} />
-          <ReadingList blockId={blockId} />
-        </div>
+      <div className="px-4 py-16 sm:px-6 lg:px-8">
+        {blockId && (
+          <div className="flex flex-col gap-8 text-base w-4/5 ml-[5%] mt-16">
+            <AddItemForm blockId={blockId} />
+            <ReadingList blockId={blockId} />
+          </div>
+        )}
       </div>
     </section>
   )
```

Comment:
Given the current implementation, which uses hooks like useState, useEffect, and useCallback, some client-side rendering is necessary. However, we can optimize this by splitting the component:
1. Create a server component for the page itself.
2. Move the client-side logic to a separate client component.
Here's how you could refactor this:

```typescript
export default function ReadingListPage() {
  return <ReadingListComponent />
}
``` 

And then move your core logic inside ReadingListComponent

App Router, page components are typically server components by defaul

Reply:
understood!